### PR TITLE
simple pagination

### PR DESCRIPTION
**Pull request by GeoffThorpeFT**
django pagination example placed into filter code to trigger pagination of rest response. Includes previous and next page links when required

### CODE REVIEW

File: cmdbv2api/views.py [hunk 1]
```diff
@@ -111,15 +114,43 @@ def getRecord(model, typeOfserializer, dictQuery):
...
+        # If page is out of range (e.g. 9999), deliver last page of results.
+        entries = paginator.page(paginator.num_pages)
 
     # define the method of extracting the fields
     serializer = typeOfserializer(entries, many=True)
 
-    return serializer.data
+    pageurl=request.build_absolute_uri(request.get_full_path()).replace('&page='+str(pagenum),'')
+    pagedoutput={}
+    if entries.has_other_pages():
```
Comment (lucas42) [inline cmdbv2api/views.py]: I don't think we should output a different result when pagination occurs.  It makes things more complicated.  When an endpoint goes above PAGE_SIZE results, everything previously using it suddenly breaks.
Also, having to access a `results` parameter is fiddly for users.  Perhaps we could add this sort of stuff in a HTTP Header?

File: cmdbv2api/views.py [hunk 2]
```diff
@@ -111,15 +114,43 @@ def getRecord(model, typeOfserializer, dictQuery):
...
 
 # Look for record/ records in table
-def filterRecords(model, typeOfserializer, dictQuery):
+def filterRecords(request, model, typeOfserializer, dictQuery):
     # define the filter to get all the records that match
 
-    entries = model.objects.filter(**dictQuery)
+    pagelimit = settings.REST_FRAMEWORK.get('PAGE_SIZE',100)
+#    rowstart = pagelimit * (pagenum - 1)
+#    rowend = pagelimit * pagenum
```
Comment (lucas42) [inline cmdbv2api/views.py]: Can you remove the commented out lines?

File: cmdbv2api/views.py [hunk 3]
```diff
@@ -111,15 +114,43 @@ def getRecord(model, typeOfserializer, dictQuery):
...
+
+    all_entries = model.objects.filter(**dictQuery) #[rowstart:rowend]
+
+    paginator = Paginator(all_entries, pagelimit) # Show 25 contacts per page
```
Comment (lucas42) [inline cmdbv2api/views.py]: I think the comment here is out of date.

File: cmdbv2api/views.py [hunk 4]
```diff
@@ -111,15 +114,43 @@ def getRecord(model, typeOfserializer, dictQuery):
...
+    pagenum = int(request.GET.get('page',1))
+    try:
+        entries = paginator.page(pagenum)
+    except PageNotAnInteger:
+        # If page is not an integer, deliver first page.
+        entries = paginator.page(1)
+    except EmptyPage:
+        # If page is out of range (e.g. 9999), deliver last page of results.
+        entries = paginator.page(paginator.num_pages)
```
Comment (lucas42) [inline cmdbv2api/views.py]: I think we should return an empty set for out of range pages.  Having the same results for multiple page numbers is confusing.

Comment (GeoffThorpeFT) [inline cmdbv2api/views.py]: I tried to use code that is close to that used by the Rest Framework (e.g. use of "results" and return of first/last page on error) - If we vary from the above then we are continuing to make it difficult to revert to pure rest framework.

Comment (lucas42) [inline cmdbv2api/views.py]: The rest framework code looks like it returns a NotFound error when you request a page out of range: https://github.com/tomchristie/django-rest-framework/blob/67ac0486f504d957fd123c8b913fb93a71cacdcd/rest_framework/pagination.py#L211 Can we follow that behaviour?

And the use of "results" seems overridable, there's a nice section here on header based pagination: http://www.django-rest-framework.org/api-guide/pagination/#header-based-pagination

File: cmdbv2api/views.py [hunk 5]
```diff
@@ -111,15 +114,55 @@ def getRecord(model, typeOfserializer, dictQuery):
...
+    # get the page required by the user
+    try:
+        pagenum = int(request.GET.get('page',1))
+        entries = paginator.page(pagenum)
+    except:
+        entries = {}
+
+    if entries:
+        # construct the prev and next links
+        pageurl=request.build_absolute_uri(request.get_full_path()).replace('&page='+str(pagenum),'')
```
Comment (lucas42) [inline cmdbv2api/views.py]: The page parameter may be the first GET parameter and hence proceeded by a `?`.

**Comment by lucas42** [comment]
Other than ending up with multiple page parameters in the links when requesting a url with `?page=`, the rest all looks great to me. :+1:

**Pull request merged**